src/core/jsdoc-tsd-parser.ts

```diff
@@ -58,7 +59,14 @@ export class JSDocTsdParser {
 				let functionBody = this.config.versionComparator.substr(this.config.versionComparator.indexOf("{") + 1);
 				functionBody = functionBody.substr(0, functionBody.length - 1).trim();
 				this.config.versionComparator = new Function("param1", "param2", functionBody);
-			} else if (fs.statSync(this.config.versionComparator).isFile()) {

```

Comment:
It's still the same problem: If the file doesn't exists, ```fs.statSync``` throws an exception. So simply revert the changes and do ```else if (fs.existsSync(this.config.versionComparator))```. Then you can do additional checks like ensure that the path is a javascript file (```path.extname(this.config.versionComparator) === "js"``` )